python/ray/air/session.py

```diff
@@ -275,6 +275,70 @@ def train_loop_per_worker():
     return session.local_rank
 
 
+@_warn_session_misuse(default_value=0)
+def get_local_world_size() -> int:
+    """Get the local rank of this worker (rank of the worker on its node).
+
+    .. code-block:: python
+
+        import time
+        from ray.air import session
+        from ray.air.config import ScalingConfig
+
+        def train_loop_per_worker():
+            if torch.cuda.is_available():
+                torch.cuda.set_device(session.get_local_rank())
+            ...
+
+        train_dataset = ray.data.from_items(
+            [{"x": x, "y": x + 1} for x in range(32)])
+        trainer = TensorflowTrainer(train_loop_per_worker,
+            scaling_config=ScalingConfig(num_workers=1),
+            datasets={"train": train_dataset})
+        trainer.fit()
+    """
+    session = _get_session()
+    if not isinstance(session, _TrainSessionImpl):
+        raise RuntimeError(
+            "`get_local_rank` can only be called for TrainSession! "
+            "Make sure you only use that in `train_loop_per_worker` function"
+            "that is passed into `DataParallelTrainer`."
+        )
+    return session.local_world_size
+
+
+@_warn_session_misuse(default_value=0)
+def get_node_rank() -> int:
+    """Get the local rank of this worker (rank of the worker on its node).
+
+    .. code-block:: python
+
+        import time
+        from ray.air import session
+        from ray.air.config import ScalingConfig
+
+        def train_loop_per_worker():
+            if torch.cuda.is_available():
+                torch.cuda.set_device(session.get_local_rank())
+            ...
+
+        train_dataset = ray.data.from_items(
+            [{"x": x, "y": x + 1} for x in range(32)])
+        trainer = TensorflowTrainer(train_loop_per_worker,
```

Comment:
the train_loop_per_worker is using torch, but we are using TensorflowTrainer here